app/models/state_change.rb

```diff
@@ -1,4 +1,6 @@
-# Performs a change of state on an asset.
+# frozen_string_literal: true
+
```

Comment:
A bit of reorganization here to group together validations/callbacks etc. I'll try and highlight the actual changes as well.